### PR TITLE
chore: consolidate eslint config

### PR DESCRIPTION
**Pull request by boneskull**
The resulting config is my best guess at the intent of the current configuration.  In addition to linting the repo (the "lint everything" commit), this change:

- Removes all `.eslintignore` , `.eslintrc` files or `package.json`-embedded config and replaces it with a single root `.eslintrc.js`
- Removes all lint-related scripts from all workspaces (including `depcheck`)
- Removes all dev deps on `eslint` and its ilk (and `ava`, since it will come from the workspace root)
- Upgrades `@metamask/eslint-config-nodejs`, `eslint`, `eslint-plugin-ava`, as well as other plugins used by specific packages; moves them all to workspace root
- Removes unused/deprecated `eslint-plugin-node` in lieu of `eslint-plugin-n`, which the new version of `@metamask/eslint-config-nodejs` needs
- Adds a prettier config (in case someone wants to use it; I'd like to add it to our workflow later)

I note that ESLint seemed to be misconfigured before this change; none of the rules specified in the root config were being checked.  This means that there are a whole lot of lint fixes that need to be made.

To lint, run `npm run lint` (which will also run `depcheck`) from the workspace root.  Likewise, `lint:fix` runs a fix (across the whole repo), and `lint:depcheck` will use lerna to run `lint:depcheck` wherever it is defined.

### CODE REVIEW

**Comment by boneskull** [comment]
I note that `depcheck` is not monorepo-aware, and depending on what we want from it, could be replaced with [eslint-plugin-import](https://npm.im/eslint-plugin-import).

**Comment by boneskull** [comment]
Worth mentioning there were multiple references to eslint rules from endo's custom config in `core/lib/strict-scope-terminator.js`.  This config doesn't appear anywhere else--I assume it was from a copy/paste--and it wasn't noticed because the aforementioned misconfiguration.

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* main
  * **PR #629** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/629" target="_blank">Graphite</a> 
    * **PR #653** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/653" target="_blank">Graphite</a> 
    * **PR #643** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/643" target="_blank">Graphite</a> 
      * **PR #620** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/620" target="_blank">Graphite</a> 
    * **PR #632** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/632" target="_blank">Graphite</a> 
    * **PR #631** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/631" target="_blank">Graphite</a> 
    * **PR #630** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/630" target="_blank">Graphite</a> 
    * **PR #627** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/627" target="_blank">Graphite</a>  👈
      * **PR #639** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/639" target="_blank">Graphite</a> 
      * **PR #617** <a href="https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/617" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/LavaMoat/LavaMoat/627?utm_source=stack-comment).

File: packages/lavapack/.depcheckrc
Comment (naugtur) [inline packages/lavapack/.depcheckrc:3]: Why is this needed?

Comment (boneskull) [reply packages/lavapack/.depcheckrc:3]: because `depcheck` is _not_ monorepo-aware, `ava` will be referenced in e.g., `lavapack/test/index.js`, but it will not appear in the `devDependencies` of `lavapack/package.json`.

any dev dep in a workspace which is required from code (unsure if this applies to dev deps only referenced in e.g., `package.json`'s `scripts` prop) but is only in the workspace root's dev deps would need to be in this `ignore` property.

**Comment by boneskull** [comment]
just gonna retarget this one to #629

**Comment by boneskull** [comment]
I'd like to merge #639 into this before merging this into `main`.

File: .eslintrc.js
Comment (legobeat) [inline .eslintrc.js]: q for later: What's the upside of `n/no-process-exit` here? Agreed on the two other.

**Review by legobeat — APPROVED** [review]
Needs a rebase before merge but LGTM otherwise!

File: .eslintrc.js
Comment (boneskull) [reply .eslintrc.js]: Not sure exactly what you're asking.  `process.exit()` can be problematic if any of your code is async, because it exits before the loop is clear.  It's safer to just let the event loop die.  Anyway; it's in the codebase, and I didn't want to remove it, so I disabled the rule for now.

**Comment by legobeat** [comment]
5e27065ef8d7f9f2adb718c4dbddd39d3675afda & 85e402f8d98479868fbe97060e08431505fa95bd should be squashed prior to merge

**Pull request merged**
